src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -356,202 +410,228 @@ private static unsafe void GetCursorPosition(out int left, out int top)
-                        TransferBytes(readBytes.Slice(readBytesPos), r);
+                        left = s_cursorLeft;
+                        top = s_cursorTop;
                         return;
                     }
 
-                    // At this point, readBytes starts with \ESC and ends with 'R'.
-                    Debug.Assert(readBytesPos > 0 && readBytesPos <= readBytes.Length);
-                    Debug.Assert(escPos == 0 && bracketPos > escPos && semiPos > bracketPos && rPos > semiPos);
-                    Debug.Assert(readBytes[escPos] == Esc);
-                    Debug.Assert(readBytes[bracketPos] == '[');
-                    Debug.Assert(readBytes[semiPos] == ';');
-                    Debug.Assert(readBytes[rPos] == 'R');
-
-                    // There are other sequences that begin with \ESC + '[' and that might be in our sequence before
-                    // the CPR, so we don't immediately trust escPos and bracketPos.  Instead, as a heuristic we trust
-                    // semiPos (which we only tracked after seeing a '[' after seeing an \ESC) and search backwards from
-                    // there looking for '[' and then \ESC.
-                    bracketPos = readBytes.Slice(0, semiPos).LastIndexOf((byte)'[');
-                    escPos = readBytes.Slice(0, bracketPos).LastIndexOf(Esc);
-
-                    // Everything before the \ESC is transferred back to the StdInReader. As is everything
-                    // between the \ESC and the '['; there really shouldn't be anything there, but we're
-                    // defensive in case the CPR wasn't written atomically and something crept in.
-                    TransferBytes(readBytes.Slice(0, escPos), r);
-                    TransferBytes(readBytes.Slice(escPos + 1, bracketPos - (escPos + 1)), r);
-
-                    // Now loop through all characters between the '[' and the ';' to compute the row,
-                    // and then between the ';' and the 'R' to compute the column. We incorporate any
-                    // digits we find, and while we shouldn't find anything else, we defensively put anything
-                    // else back into the StdInReader.
-                    ReadRowOrCol(bracketPos, semiPos, r, readBytes, ref top);
-                    ReadRowOrCol(semiPos, rPos, r, readBytes, ref left);
-
-                    // Mark that we've successfully received a CPR response at least once.
-                    s_everReceivedCursorPositionResponse = true;
-                }
-                finally
-                {
-                    Interop.Sys.UninitializeConsoleAfterRead();
-                    s_firstCursorPositionRequest = false;
+                    cursorVersion = s_cursorVersion;
                 }
 
-
-                bool BufferUntil(byte toFind, StdInReader src, ref Span<byte> dst, ref int dstPos, out int foundPos)
+                // Create a buffer to read the response into.  We start with stack memory and grow
```

Comment:
GitHub shows a lot of changes here. This is mostly indentation.

Reply:
> GitHub shows a lot of changes here. This is mostly indentation.

Thanks.  It's much easier to read with:
![image](https://user-images.githubusercontent.com/2642209/54375675-e2a3de80-4657-11e9-8a54-ea767be0a013.png)

Reply:
Good to learn that exists.